Update Workflow State icons

kbrock opened the pull request:
Before
======

Icons were working for Tasks
Icons were only displaying play for all others

<img width="359" alt="Monosnap ManageIQ: Requests 2024-06-04 17-00-26" src="https://github.com/ManageIQ/manageiq-ui-classic/assets/1930/e9be9833-5539-4381-ab6e-cf4fc2826f62">

After
=====

Icons are displaying for all Tasks

<img width="359" alt="Monosnap ManageIQ: Requests 2024-06-05 09-43-55" src="https://github.com/ManageIQ/manageiq-ui-classic/assets/1930/a366e01c-e76f-478f-b266-82db9b713606">

Fryguy commented on app/javascript/components/request-workflow-status/data.js:
```diff
@@ -65,12 +65,21 @@ const convertDuration = (duration) => {
+  if (item.FinishedTime) {
+    if (item.Output && item.Output.Error) {
+      if (item.RetryCount) {
+        // TODO:
```
What are the TODO for?

Fryguy commented on app/javascript/components/request-workflow-status/data.js:
```diff
@@ -65,12 +65,21 @@ const convertDuration = (duration) => {
+        return { icon: 'carbon--MisuseOutline' }
+      }
+    } else {
+      return { icon: 'carbon--CheckmarkOutline', tooltip: 'yay' }
```
Don't think we need this tooltip.

Fryguy commented on app/javascript/components/request-workflow-status/data.js:
```diff
@@ -65,12 +65,21 @@ const convertDuration = (duration) => {
+    if (item.Output && item.Output.Error) {
+      if (item.RetryCount) {
+        // TODO:
+        return { icon: 'carbon--RetryFailed', tooltip: `${item.Output.Error} : ${item.RetryCount}` }
```
tooltip wise, I'm concerned the error will be too big for a tooltip.  If there is no error, will it show ` : 4`?  Also there's no indicator that the trailing number is the retry count.

Fryguy commented:
@kbrock Can we take out the tooltip stuff and do that in a separate PR?  I think fixing the icons can be it's own thing.

Fryguy commented:
Again, can we remove the tooltips / titles entirely from this PR?  I'd like to merge the bug fix separate from the enhancement.

I'm good with the new icon for retry failed.

kbrock commented:
@agrare we were looking at Task completion instead of at the state level
@Fryguy Thanks to @GilbertCherrie all is set with tooltops.

update:
- removed `s/yay/boo/g`
- added tooltips
- removed TODOs

update:
- fixed color for retry
- removed tooltips

---

Dev notes:

```
        let title = item.Output.Cause ? `${item.Output.Error} : ${item.Output.Cause}` : item.Output.Error;
        if (title.length > 100) {
          title = title.substring(0, 100)
        }
        const title = sprintf(__(`Retry Count: %d`), item.RetryCount);
```

kbrock commented:
wip: I need to test the just failing case.

Think the retry logic should be bubbled up one level.
Quickly checking and will be all set to merge soon

kbrock commented:
update:
- fixed: failures now display as failure (was success)
- fixed: retry successes now displays as success (was displaying as a failure)

un-wip: resolved

Fryguy commented:
Backported to `radjabov` in commit 0301d74f503a42cd2f730a3d42c60ab0ebcced15.

```text
commit 0301d74f503a42cd2f730a3d42c60ab0ebcced15
Author: Adam Grare <adam@grare.com>
Date:   Wed Jun 5 10:28:14 2024 -0400

    Merge pull request #9200 from kbrock/state_logos
    
    Update Workflow State icons
    
    (cherry picked from commit e5d7cb22689b581662a09917a24006167f247a4c)
```

jeffbonson commented on app/javascript/components/request-workflow-status/data.js:
```diff
@@ -65,12 +65,19 @@ const convertDuration = (duration) => {
-    return { icon: 'carbon--CheckmarkOutline' };
-  } if (item.RunnerContext && item.RunnerContext.Error) {
-    return { icon: 'carbon--MisuseOutline' };
+  if (item.FinishedTime) {
+    if (item.Output && item.Output.Error) {
+      if (item.RetryCount) {
+        return { icon: 'carbon--RetryFailed' }
+      } else {
+        return { icon: 'carbon--MisuseOutline' }
+      }
+    } else {
+      return { icon: 'carbon--CheckmarkOutline' }
+    }
+  } else {
+    return { icon: 'carbon--PlayOutline' }
```
Just putting it over here to reduce a few lines later - 

```
const getItemIcon = (item) => {
  if (item.FinishedTime) {
    if (item.Output && item.Output.Error) {
      return { icon: item.RetryCount ? 'carbon--RetryFailed' : 'carbon--MisuseOutline' };
    }
    return { icon: 'carbon--CheckmarkOutline' };
  }
  return { icon: 'carbon--PlayOutline' };
};
```
